target_redshift/db_sync.py

```diff
@@ -653,109 +646,81 @@ def load_csv(self, s3_key, count, size_bytes, compression=False, manifest=False)
                             for pkey in names
                         ]
                     )
-                    sql = """
+                    sql = run(
+                        f"""
                     DELETE FROM {stage_table}
                     USING 
                       (SELECT {pkeys}, max(_sdc_sequence) AS "_sdc_sequence" FROM {stage_table} s GROUP BY {pkeys}) ss 
                     WHERE {join_condition} AND ss._sdc_sequence <> {stage_table}._sdc_sequence
-                    """.format(
-                        pkeys=pkeys,
-                        stage_table=stage_table,
-                        join_condition=join_condition,
+                    """
                     )
-                    self.logger.info("Running query: {}".format(sql))
-                    cur.execute(sql)
 
                     if not self.skip_updates:
-                        update_sql = """
+                        join_condition = " AND ".join(
+                            ["stage.{c} = target.{c}".format(c=pkey) for pkey in names]
+                        )
+
+                        run(
+                            f"""
                             UPDATE {target_table}
-                            SET _sys_end_time = stage._sys_updated_at
+                            SET _sys_end_time = COALESCE(stage._sys_updated_at, stage._sdc_deleted_at)
```

Comment:
Part of the problem is this. This is more of an extra case to make sure we are marking things to delete.

Reply:
Do you expect `_sys_updated_at` to be null?

Reply:
I don't expect _sys_updated_at to be null ever. I'll take it out because I prefer simplicity myself